### PR TITLE
Add a funding widget in the funding page

### PR DESCRIPTION
**Pull request by leoapost**
### Trello card
https://trello.com/c/NE9ALBAW

### Context
Currently, the users need to go through the funding page and read a lot of text to see what funding options they are eligible for the subject they are interested at.

Adding this simple widget at the top of the funding page, allows them to quickly see their funding options by simply choosing a subject. 

### Changes proposed in this pull request
Create a view component to render a funding widget with a form to look up funding options per subject and display the results. The responsible controller method (pages#show) is not meant to support such interactivity, so this is not the most elegant implementation.  

This PR also includes Capybara configurations for running feature tests with `js: true` on headless chrome/chromium. The CI will be using chromium which has smaller footprint, but had to replace the `webdrivers` gem with the `selenium-webdriver` gem because the drivers the `webdrivers` gem is downloading are not working very well in Alpine.

### Guidance to review
| Before | After |
| --- | --- |
|funding-page-before|funding-page-after|

### CODE REVIEW

File: app/controllers/pages_controller.rb
Comment (ethax-ross) [inline app/controllers/pages_controller.rb:40]: I would maybe call this out into its own route, feels wrong having it in the generic `pages#show` action

File: app/models/funding_widget.rb
Comment (ethax-ross) [inline app/models/funding_widget.rb:5]: Worth moving this to the translations file so its not polluting the model, then you can just have a set of keys here?

File: app/components/funding_widget_component.rb
Comment (ethax-ross) [inline app/components/funding_widget_component.rb:12]: Found this quite hard to follow... not sure if this is any better/if its right, but:

```
form.subjects.index_by { |_, v| v[:education] }.transform_values { |v| v.first }
```

File: app/views/sections/_head.html.erb
Comment (ethax-ross) [inline app/views/sections/_head.html.erb:5]: Can this go in to `/javascript` instead of being inline?

File: app/webpacker/styles/layout.scss
Comment (ethax-ross) [inline app/webpacker/styles/layout.scss:34]: Will changing this selector alter the behaviour of any other controllers that toggle this `hidden` class? (they may have their own selectors dealing with JS enabled/disabled?)

**Review by ethax-ross — COMMENTED** [review]
Looks great, just a couple of comments/suggestions.

File: app/webpacker/styles/layout.scss
Comment (leoapost) [reply app/webpacker/styles/layout.scss:34]: No this is safe, it will not change the behaviour of any other controllers.

File: app/components/funding_widget_component.rb
Comment (leoapost) [reply app/components/funding_widget_component.rb]: I've refactored this into an array and extracted it to smaller methods to make it simpler to understand.

**Review by ethax-ross — APPROVED** [review]
LGTM

**Review by ethax-ross — DISMISSED** [review]

**Comment by MylesJarvis** [comment]
To ship Monday 10 May

**Review by ethax-ross — DISMISSED** [review]

**Comment by MylesJarvis** [comment]
Now awaiting further review from policy colleagues

**Review by MylesJarvis — DISMISSED** [review]
Testing approving it to see if review app builds - NOT TO MERGE

File: app/views/content/funding-your-training.md
Comment (MylesJarvis) [inline app/views/content/funding-your-training.md:26]: @peteryates here's the component in need of an <a href="tel:08003892500" class="telephone-number"
    aria-label="Telephone">0800 389 2500</a>

**Comment by MylesJarvis** [comment]
@peteryates are you okay to resolve conflicts too?

**Review by peteryates — APPROVED** [review]
All looks fine to me.

The addition of JavaScript testing from our specs is an interesting one - the overlap between internal tests and the Cypress ones is increasing.

I'm not against it, just wonder whether we should look at combining them at some point.

**Review by MylesJarvis — APPROVED** [review]
LGTM

**Pull request merged**
